### PR TITLE
[Optimize] Flush batches when reindexing blocks instead of commiting every transaction

### LINKED ISSUE #1096: Speed up wsv restore procedure from block store
When block indexer indexes block we do flush (block indexer impl), which performs commit.
This is not optimal thing as we might have millions of blocks and flushing every time could be slow

### PR DESCRIPTION
**Pull request by kuvaldini**
Closes #1096
See also #1109

### CODE REVIEW

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp [hunk 1]
```diff
@@ -136,7 +137,13 @@ namespace {
         }
       }
 
-      if (not validator.validateAndApply(std::move(block), *mutable_storage)) {
+      // flush every 1000 blocks
+      //      int reindex_blocks_flush_cache_size_in_blocks = 10;
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/wsv_restorer_impl.cpp:141]: Uncomment or remove

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp [hunk 2]
```diff
@@ -136,7 +137,13 @@ namespace {
         }
       }
 
-      if (not validator.validateAndApply(std::move(block), *mutable_storage)) {
+      // flush every 1000 blocks
```
Comment (kamilsa) [inline irohad/ametsuchi/impl/wsv_restorer_impl.cpp:140]: I think not 1000, but reindex_blocks_flush_cache_size_in_blocks_ blocks

**Review by kamilsa — APPROVED** [review]

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp (hunk 2 above)
Comment (kuvaldini) [reply irohad/ametsuchi/impl/wsv_restorer_impl.cpp:140]: yeah fixed

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp (hunk 1 above)
Comment (kuvaldini) [reply irohad/ametsuchi/impl/wsv_restorer_impl.cpp:141]: fixed

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp [hunk 3]
```diff
@@ -136,7 +137,12 @@ namespace {
         }
       }
 
-      if (not validator.validateAndApply(std::move(block), *mutable_storage)) {
+      // flush every N blocks
+      auto counter = height - starting_height + 1;
+      bool do_flash = counter % reindex_blocks_flush_cache_size_in_blocks_ == 0
```
Comment (iceseer) [inline irohad/ametsuchi/impl/wsv_restorer_impl.cpp:142]: ```bool const```

File: irohad/main/iroha_conf_loader.hpp [hunk 4]
```diff
@@ -89,6 +89,7 @@ struct IrohadConfig {
   boost::optional<Crypto> crypto;
 
   std::string metrics_addr_port;
+  int reindex_blocks_flush_cache_size_in_blocks = 1000;
```
Comment (iceseer) [inline irohad/main/iroha_conf_loader.hpp:92]: ```uint32_t```

**Review by iceseer — APPROVED** [review]

File: irohad/ametsuchi/impl/wsv_restorer_impl.cpp (hunk 3 above)
Comment (kuvaldini) [reply irohad/ametsuchi/impl/wsv_restorer_impl.cpp]: doen

File: irohad/main/iroha_conf_loader.hpp (hunk 4 above)
Comment (kuvaldini) [reply irohad/main/iroha_conf_loader.hpp]: `unsigned` everywhere

Comment (kuvaldini) [reply irohad/main/iroha_conf_loader.hpp]: done

**Pull request closed without merging**